pkg/performanceprofile/controller/performanceprofile/components/kubeletconfig/kubeletconfig_test.go

```diff
@@ -22,7 +22,7 @@ const testReservedMemory = `reservedMemory:
 var _ = Describe("Kubelet Config", func() {
 	It("should generate yaml with expected parameters", func() {
 		profile := testutils.NewPerformanceProfile("test")
-		selectorKey, selectorValue := components.GetFirstKeyAndValue(profile.Spec.MachineConfigPoolSelector)
+		selectorKey, selectorValue := components.GetFirstKeyAndValue(profile.Spec.NodeSelector)
 		kc, err := New(profile, &components.KubeletConfigOptions{MachineConfigPoolSelector: map[string]string{selectorKey: selectorValue}, MixedCPUsEnabled: true})
```

Comment:
This will not work. Because even though NTO will figure things out, you are not calling the proper method here. Instead you just directly pass the node selector to a KubeletConfig.

Reply:
The controller does it like this:
- MCP is detected here https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile_controller.go#L462
- MCP selector is computed by https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile/components/profile/profile.go#L11
- And passed into the KubeletConfig like this: https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile/components/manifestset/manifestset.go#L63

Reply:
@MarSik so IIUC, although the tests are passing, the whole PR is not needed?

Reply:
The tests are passing because you only check that the passed in values are reflected in the generated yaml. And that happens. But the semantics is wrong.

About the need.. it depends on what we want to test here. If string manipulation is the goal, then we do not need the patch.

But maybe we should make sure the MCP autodetection works properly and then it needs to be improved to use it.

Reply:
Got you, then I'm closing the PR